common/preformance.py

```
@@ -0,0 +1,1250 @@
+    Optimized list serializer for bulk operations with query optimization.
+    """
+
+    def to_representation(self, data):
+        """
+        Override to apply bulk optimizations to the queryset.
+        """
+        # Apply optimizations if data is a queryset
+        if hasattr(data, "select_related") and hasattr(
+            self.child, "select_related_fields"
+        ):
+            if self.child.select_related_fields:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)